src/components/scenes/EdgeLoginScene.tsx

```diff
@@ -1,242 +1,158 @@
+    navigation.navigate('waitScene', { message: s.strings.edge_login_fetching_message })
+    dispatch(lobbyLogin()).then(() => {
+      navigation.goBack()
+    })
+  })
 
-export class EdgeLoginSceneComponent extends React.Component<Props> {
-  renderBody() {
-    let message = this.props.error
-    if (!this.props.error) {
+  const renderBody = useHandler(() => {
+    let message = error ?? ''
+
+    if (!error) {
       message = sprintf(s.strings.access_wallet_description, config.appName)
     }
-    if (!this.props.lobby && !this.props.error) {
+
+    if (!lobby && !error) {
       throw new Error('Not normal expected behavior')
     }
-    if (this.props.lobby && this.props.lobby.loginRequest && this.props.lobby.loginRequest.appId === '') {
-      message = sprintf(s.strings.edge_description_warning, this.props.lobby.loginRequest.displayName)
+    if (lobby && lobby.loginRequest && lobby.loginRequest.appId === '') {
+      message = sprintf(s.strings.edge_description_warning, lobby.loginRequest.displayName)
     }
     return (
-      <View style={styles.body}>
-        <Text style={styles.bodyText}>{message}</Text>
+      <View style={styles.warning}>
+        <Alert title={s.strings.string_warning} message={message} type="warning" numberOfLines={6} />
       </View>
     )
-  }
+  })
 
-  renderButtons() {
-    const { navigation } = this.props
+  const renderButtons = useHandler(() => {
     const handleDecline = () => navigation.goBack()
-
-    if (this.props.isProcessing) {
+    if (isProcessing) {
       return (
         <View style={styles.buttonsProcessing}>
-          <ActivityIndicator color={THEME.COLORS.ACCENT_MINT} />
+          <ActivityIndicator color={theme.primaryText} size="large" />
         </View>
       )
     }
-    if (this.props.error) {
+    if (error) {
       return (
-        <View style={styles.buttonContainer}>
-          <View style={styles.buttons}>
-            <SecondaryButton style={styles.cancelSolo} onPress={handleDecline}>
-              <SecondaryButton.Text>{s.strings.string_cancel_cap}</SecondaryButton.Text>
-            </SecondaryButton>
-          </View>
+        <View style={styles.buttons}>
+          <MainButton label={s.strings.string_cancel_cap} type="escape" onPress={handleDecline} />
         </View>
       )
     }
     return (
-      <View style={styles.buttonContainer}>
-        <View style={styles.buttons}>
-          <SecondaryButton style={styles.cancel} onPress={handleDecline}>
-            <SecondaryButton.Text>{s.strings.string_cancel_cap}</SecondaryButton.Text>
-          </SecondaryButton>
-          <PrimaryButton style={styles.submit} onPress={this.props.accept}>
-            <PrimaryButton.Text>{s.strings.accept_button_text}</PrimaryButton.Text>
-          </PrimaryButton>
-        </View>
+      <View style={styles.buttons}>
+        <MainButton marginRem={[1, 0, 1, 0]} label={s.strings.accept_button_text} onPress={handlePress} />
```

Comment:
If you pass an async `handlePress` to `MainButton`, it will show a spinner while it does the work. This may be simpler & better looking than the big `if` statement you have here to manually show a spinner.